submissions/ivan-chukhalo/frogger-game/app.js

```diff
@@ -0,0 +1,121 @@
+const AVAILABLE_ROWS_FOR_ENEMIES = [63, 146, 229]; 
+const allEnemies = [];
+
+const WIDTH_OF_GAME_SPRITES = 100;
+const COLUMNS_ROWS_ON_GAME_FIELD = 5;
+const HEIGH_OF_GAME_FIELD_CELL = 83;
+const PLAYER_BUG_CENTERS_DISTANCE_OF_TOUCHING = 80;
+const MIN_SPEED_MULTIPLIER_FOR_BUG = 200;
+const MAX_SPEED_MULTIPLIER_FOR_BUG = 350;
+const Y_OF_HIGHEST_ROW = 63;
+const Y_OF_GAME_FIELD_ROWS = [Y_OF_HIGHEST_ROW];
+(function calculating_coords_of_game_rows(){
+    for (let i = 1; i < 5; i++){
+        Y_OF_GAME_FIELD_ROWS.push(Y_OF_HIGHEST_ROW + HEIGH_OF_GAME_FIELD_CELL * i);
+    }
+})();
+const Y_OF_LOWEST_GAME_FIELD_ROW = Y_OF_GAME_FIELD_ROWS[4];
+
+function resetPlayer(){
+    player.x = player.xInitial;
+    player.y = player.yInitial;
+}
+
+function resetEnemies(){
+    allEnemies.forEach((el)=>{        
+        el.x = el.xInitial;
+        el.speedKoef = Math.floor(Math.random() * MAX_SPEED_MULTIPLIER_FOR_BUG - MIN_SPEED_MULTIPLIER_FOR_BUG) + MIN_SPEED_MULTIPLIER_FOR_BUG;  
+    })
+}
+
+function isOnWater(){                   
+    if (this.y === this.yAxisMin){
+        resetEnemies();
+        resetPlayer();
+    }
+}
+
+
+var Enemy = function(y) {                    
+    this.sprite = 'images/enemy-bug.png';
+    this.speedKoef = Math.floor(Math.random() * MAX_SPEED_MULTIPLIER_FOR_BUG - MIN_SPEED_MULTIPLIER_FOR_BUG) + MIN_SPEED_MULTIPLIER_FOR_BUG;  
+    this.xInitial = -WIDTH_OF_GAME_SPRITES;
+    this.xAxisMin = this.xInitial;
+    this.x = this.xInitial;
+    this.xAxisMax = WIDTH_OF_GAME_SPRITES * COLUMNS_ROWS_ON_GAME_FIELD;
+    this.y = y;
+};
+
+Enemy.prototype.doesHitPlayer = function (input_player){
+    if (Math.abs(this.x - input_player.x) <= PLAYER_BUG_CENTERS_DISTANCE_OF_TOUCHING && this.y === input_player.y){   
+        resetPlayer();
+        resetEnemies();
+    }
+}
+
+Enemy.prototype.update = function(dt) {     
+    this.x = this.x >= this.xAxisMax ? this.x = this.xInitial : this.x + dt * this.speedKoef;
+    this.doesHitPlayer(player);   
```

Comment:
`player` is still a global variable.
The requirements provide a hint on how to resolve the problem.

Reply:
@OleksiyRudenko, excuse me.

It seems like I've missed this row or I got rid of global variables dependencies in the project, then made some extra changes in the code and added new dependency without noticing it)

Fixed issue. Review, please.

Reply:
This line of code seems to remain unchanged. `player` is a global variable.